lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -705,6 +705,10 @@ actor ConnectorSink is Sink
     Serialize hard state (i.e., can't afford to lose it) and send
     it to the local event log and reset 2PC state.
     """
+    if _twopc.state_is_2abort() then
```

Comment:
I think the intention is clearer here if you never call `checkpoint_state()` under this condition in the first place.

Reply:
Fixed by d0f9c924dac